core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -740,6 +743,7 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
+    successfulTaskDurations.remove(taskInfos(tid).duration)
```

Comment:
so can't we just change the line below back to `if (tasksSuccessful >= minFinishedForSpeculation && tasksSuccessful > 0) {` and still get rid of the call to `remove`?